dask_glm/logistic.py

```diff
@@ -220,6 +220,32 @@ def newton(X, y, max_iter=50, tol=1e-8):
     return beta
 
 
+def compute_prox_step(beta, gradient, func, X, y, prox, stepSize=1.0):
+    '''Computes stepsize for proximal gradient method using
+    backtracking line search.'''
+
+    backtrackMult = 0.1
+
+    # Compute the step size
+    obeta = beta
+    lf = func
+    for ii in range(100):
+        beta = prox(obeta - stepSize * gradient, stepSize)
+        step = obeta - beta
+        Xbeta = X.dot(beta).compute()  # ugh
```

Comment:
Unfortunately not; the beta update in proximal gradient descent is non-linear, so the new `Xbeta` can't be computed purely from `obeta` and `Xstep` (at least not that I am aware of).  Maybe @mcg1969 has some input about the best way to do this, avoiding repeated matrix-vector operations within the line search?